src/modules/lua_mtev.c

```diff
@@ -5513,6 +5513,44 @@ nl_getaddrinfo(lua_State *L) {
   return 2;
 }
 
+/*! \lua rc, family, addr = mtev.inet_pton(address)
+\brief Wrapper around inet_pton(3). Can be used to validate IP addresses and detect the address family (IPv4,IPv6)
+\param address to parse
+\return rc true if address is a valid IP address, false otherwise
+\return family address family of the address, either "inet" or "inet6".
+\return addr struct in_addr as udata
+
+*/
+
+static int
+nl_inet_pton(lua_State *L) {
+  if(lua_gettop(L) != 1) {
+    return luaL_error(L, "mtev.inet_pton expects exactly one argument");
+  }
+  const char *host = luaL_checkstring(L, 1);
+  struct in_addr addr;
+  int rc;
+  rc = inet_pton(AF_INET, host, &addr);
+  if (rc == 1) {
+    lua_pushboolean(L, 1);
+    lua_pushstring(L, "inet");
+    struct in_addr *u = lua_newuserdata(L, sizeof(struct in_addr));
+    *u = addr;
+    return 3;
+  }
+  rc = inet_pton(AF_INET6, host, &addr);
+  if (rc == 1) {
+    lua_pushboolean(L, 1);
+    lua_pushstring(L, "inet6");
+    struct in_addr *u = lua_newuserdata(L, sizeof(struct in_addr));
```

Comment:
this isn't an in_addr... it's an `struct in6_addr`

Reply:
Uh. The manual was not exactly transparent about this:

>     The inet_pton() function converts a presentation format address (that is,
>     printable form as held in a character string) to network format (**usually
>     a struct in_addr** or some other internal binary representation, in network
>     byte order).

Where is the return type documented?